dot/core/service.go

```diff
@@ -579,3 +609,33 @@ func (s *Service) GetMetadata(bhash *common.Hash) ([]byte, error) {
 	s.rt.SetContextStorage(ts)
 	return s.rt.Metadata()
 }
+
+// RegisterRuntimeUpdatedChannel function to register chan that is notified when runtime version changes
+func (s *Service) RegisterRuntimeUpdatedChannel(ch chan<- runtime.Version) (byte, error) {
+	s.runtimeChangedLock.RLock()
+
+	if len(s.runtimeChanged) == 256 {
+		return 0, errors.New("channel limit reached")
+	}
+
+	var id byte
+	for {
+		id = generateID()
+		if s.runtimeChanged[id] == nil {
+			break
+		}
+	}
+
+	s.runtimeChangedLock.RUnlock()
+
+	s.runtimeChangedLock.Lock()
+	s.runtimeChanged[id] = ch
+	s.runtimeChangedLock.Unlock()
```

Comment:
why are needed these two locks? Just `s.runtimeChangedLock.Lock()` could not be enough?

Reply:
Right, I've updated to use just one lock (and unlock with defer).